### PR TITLE
Add cert authorization with common name support.

### PR DESCRIPTION
**Pull request by liuh-80**
Add cert authorization with common name support.

#### Why I did it
Support cert authorization with common name.

#### How I did it
Load trusted cert common name from config DB and check cert common name. 

#### How to verify it
Manually test.
Add new UT.

#### Work item tracking
Microsoft ADO (number only): 25226269

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106
- [ ] 202111

#### Description for the changelog
Add cert authorization with common name support.

### CODE REVIEW

**Comment by liuh-80** [comment]
The yang model and service start script change in this PR: https://github.com/sonic-net/sonic-buildimage/pull/18709
Will create test case PR after this PR merged.

File: dialout/dialout_server_cli/dialout_server_cli.go [hunk 1]
```diff
@@ -22,6 +22,7 @@ var (
 	serverKey         = flag.String("server_key", "", "TLS server private key")
 	insecure          = flag.Bool("insecure", false, "Skip providing TLS cert and key, for testing only!")
 	allowNoClientCert = flag.Bool("allow_no_client_auth", false, "When set, telemetry server will request but not require a client certificate.")
+	clientCrtCname    = flag.String("client_crt_cname", "", "Client cert common name")
```
Comment (ganglyu) [inline dialout/dialout_server_cli/dialout_server_cli.go:25]: Do we need cname for dialout server?

Comment (liuh-80) [inline dialout/dialout_server_cli/dialout_server_cli.go:25]: WIll fix this later.

File: gnmi_server/server_test.go [hunk 2]
```diff
@@ -4187,6 +4193,187 @@ func TestSaveOnSet(t *testing.T) {
...
+    defer mockpopulate.Reset()
+
+    // check auth with nil cert name
+    ctx, cancel := CreateAuthorizationCtx()
+    ctx, err = ClientCertAuthenAndAuthor(ctx, "")
+    if err != nil {
+        t.Errorf("CommonNameMatch with empty config table should success: %v", err)
+    }
+
+	cancel()
```
Comment (ganglyu) [inline gnmi_server/server_test.go:4250]: Not aligned.

File: dialout/dialout_server_cli/dialout_server_cli.go (hunk 1 above)
Comment (liuh-80) [reply dialout/dialout_server_cli/dialout_server_cli.go:25]: Will check with Zain

File: gnmi_server/server_test.go (hunk 2 above)
Comment (liuh-80) [reply gnmi_server/server_test.go:4250]: Fixed this and similar case

**Review by ganglyu — APPROVED** [review]

**Comment by bingwang-ms** [comment]
Cherry-pick to 202405 is done by PR #322

**Pull request merged**